src/fusion_power_plant.cc

```diff
@@ -10,14 +10,98 @@ std::string FusionPowerPlant::str() {
   return Facility::str();
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void FusionPowerPlant::EnterNotify() {
+  //Pseudocode implementation of EnterNotify()
+  cyclus::Facility::EnterNotify();
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  //Truncated for readability. It goes on Tritium Storage and gets started, though.
+  fuel_startup_policy.Init(&tritium_storage).Set(fuel_incommod).Start();
+  
+  //Truncated for readability
+  blanket_fill_policy.Init(&blanket).Set(blanket_incommod).Start();
+
+  //Tritium Buy Policy Section:
+  if (refuel_mode == "schedule") {
+    //Set up a schedule fill policy with Katie's active/dormant policy code
+    //(not shown here). Do not start the policy yet.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    //otherwise we just do it normally, but still don't start it.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else {
+    //This can be more sophisticated later.
+    throw cyclus::KeyError("Refill mode not recognized!");
+  }
+
+  tritium_sell_policy.Init(&tritium_excess).Set(fuel_incommod).Start();
+  helium_sell_policy.Init(&helium_storage).Set(he3).Start();
+
+}
+
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tick() {
-  std::cout<<"Hello"<<std::endl;
+  //pseudocode implementation of Tick():
+  
+  if (can_operate) {
+    SequesterTritium();
+    OperateReactor();
+  }
+
+  //I think somehow combining all these would be nice...
+  DecayInventory(tritium_storage);
+  ExtractHelium(tritium_storage);
+
+  DecayInventory(tritium_excess);
+  ExtractHelium(tritium_excess);
+
+  sequestered_tritium->Decay(context()->time());
+
+  //This was a bunch of free-floating code earlier, but i've truncated it 
+  //because it's mostly for DRE stuff.
+  MoveExcessTritiumToSellBuffer();
+
+  //This maybe belongs in its own function?
+  if (context()->time() % blanket_turnover_frequency == 0 && !blanket.empty()) {
+    if (blanket.quantity() >= blanket_turnover) {
+      blanket_excess.Push(blanket.Pop(blanket_turnover));
+      CombineInventory(blanket_excess);
+      RecordOperationalInfo("Blanket Cycled");
+    } else {
+      RecordOperationalInfo("Blanket Not Cycled");
+    }
+  }
+  
+
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tock() {
-  std::cout<<"There!"<<std::endl;
+  //pseudocode implementation of Tock():
+  
+  if(!can_operate) {
+    try { 
+      CheckOperatingConditions(); //Previously "Startup()"
+      fuel_startup_policy.Stop();
+      fuel_refill_policy.Start();
```

Comment:
We should discuss whether we can come up with a better buy policy that doesn't require this

Reply:
Can you make a comment about what the difference is between these policies?

Reply:
Or maybe a function pseudocode that describes it

Reply:
fuel_startup_policy is an (s,S) policy (to use Katie's lingo) which demands the reactor be filled to some user-determined startup fuel inventory whenever it's below that amount (for instance, buy any amount of tritium up to a total inventory of 5kg while below 5kg), whereas the idea with the fuel_refill_policy was that it could be a number of types of buying policies. I implemented two: a "fill" strategy which is basically identical to the startup policy, and a "schedule" policy which buys a fixed amount every few timesteps determined by the user (for instance, once every 6 months it buys 1 kg). The idea was that more could be implemented in the future, since this was intended as a first pass at the agent, but that basically this is where we switch from the buying strategy to get going initially to operating.